### PR TITLE
Remove multi_json  as dependency in favor of std-lib json

### PR DESCRIPTION
**Pull request by otherguy**
This PR is almost identical to #212 but for security reasons we want to maintain our own fork.

CI is also updated to include newer Ruby and Rails versions (currently failing on `edge` but that is expected).

### CODE REVIEW

File: lib/active_record/session_store.rb [hunk 1]
```diff
@@ -62,12 +62,12 @@ def self.dump(value)
       # Uses built-in JSON library to encode/decode session
       class JsonSerializer
         def self.load(value)
-          hash = MultiJson.load(value)
+          hash = JSON.load(value)
```
[the remark this answers is not in the document]
Comment (otherguy) [reply lib/active_record/session_store.rb]: @byroot that is an excellent point and I didn't even notice this 😱 Let me replace that real quick.

**Comment by byroot** [comment]
Looks fine, but lots of issues with CI. Let's get the MultiJson removal and CI fixes in different PRs.

**Comment by byroot** [comment]
I merged the dependency removal. 

However I tried to really update the CI matrix but it's a huge pain -_-.

**Comment by otherguy** [comment]
@byroot thank you for resolving this with https://github.com/rails/activerecord-session_store/commit/536716a98a85a46a320bc5da020d9dd37174112c :)

As for the CI matrix, with how frequent commits to `rails/rails` are, I have excluded `edge` from my matrix. Perhaps you can add `7.2` and make failures of `edge` just warnings?

**Comment by byroot** [comment]
The failure on edge is also on 7.2. I dug a bit as to what is going on, but I'm a bit tired.

**Comment by otherguy** [comment]
@byroot can I help you get this released as a new version? What's missing?

**Comment by byroot** [comment]
The CI would need to be fixed, tests are no longer passing against Rails edge because of some change in the router.

I know the gem works fine, it's just a test helper that was reaching into Action Dispatch internals that need to be refactored/updated.
